libraries/Filter/AP_Filter.cpp

```diff
@@ -0,0 +1,152 @@
+    AP_SUBGROUPINFO(params[2], "3_", 6, AP_Filters, AP_Filter_params),
+    // @Group: 3_
+    // @Path: AP_NotchFilter_params.cpp
+    AP_SUBGROUPVARPTR(filters[2], "3_", 7, AP_Filters, backend_var_info[2]),
+#endif
+#if AP_FILTER_NUM_FILTERS >= 4
+    // @Group: 4_
+    // @Path: AP_Filter_params.cpp
+    AP_SUBGROUPINFO(params[3], "4_", 8, AP_Filters, AP_Filter_params),
+    // @Group: 4_
+    // @Path: AP_NotchFilter_params.cpp
+    AP_SUBGROUPVARPTR(filters[3], "4_", 9, AP_Filters, backend_var_info[3]),
+#endif
+#if AP_FILTER_NUM_FILTERS >= 5
+    // @Group: 5_
+    // @Path: AP_Filter_params.cpp
+    AP_SUBGROUPINFO(params[4], "5_", 10, AP_Filters, AP_Filter_params),
+    // @Group: 5_
+    // @Path: AP_NotchFilter_params.cpp
+    AP_SUBGROUPVARPTR(filters[4], "5_", 11, AP_Filters, backend_var_info[4]),
+#endif
+#if AP_FILTER_NUM_FILTERS >= 6
+    // @Group: 6_
+    // @Path: AP_Filter_params.cpp
+    AP_SUBGROUPINFO(params[5], "6_", 12, AP_Filters, AP_Filter_params),
+    // @Group: 6_
+    // @Path: AP_NotchFilter_params.cpp
+    AP_SUBGROUPVARPTR(filters[5], "6_", 13, AP_Filters, backend_var_info[5]),
+#endif
+#if AP_FILTER_NUM_FILTERS >= 7
+    // @Group: 7_
+    // @Path: AP_Filter_params.cpp
+    AP_SUBGROUPINFO(params[6], "7_", 14, AP_Filters, AP_Filter_params),
+    // @Group: 7_
+    // @Path: AP_NotchFilter_params.cpp
+    AP_SUBGROUPVARPTR(filters[6], "7_", 15, AP_Filters, backend_var_info[6]),
+#endif
+#if AP_FILTER_NUM_FILTERS >= 8
+    // @Group: 8_
+    // @Path: AP_Filter_params.cpp
+    AP_SUBGROUPINFO(params[7], "8_", 16, AP_Filters, AP_Filter_params),
+    // @Group: 8_
+    // @Path: AP_NotchFilter_params.cpp
+    AP_SUBGROUPVARPTR(filters[7], "8_", 17, AP_Filters, backend_var_info[7]),
+#endif
+    AP_GROUPEND
+};
+
+const AP_Param::GroupInfo *AP_Filters::backend_var_info[AP_FILTER_NUM_FILTERS];
+
+AP_Filter::AP_Filter(FilterType type):
+    _type(type)
+{
+}
+
+AP_Filters::AP_Filters()
+{
+    AP_Param::setup_object_defaults(this, var_info);
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (singleton != nullptr) {
+        AP_HAL::panic("AP_Filters must be singleton");
+    }
+#endif
+    singleton = this;
+}
+
+void AP_Filters::init()
+{
+    uint8_t count = 0;
+    // make sure all custom rotations are allocated
+    for (uint8_t i = 0; i < AP_FILTER_NUM_FILTERS; i++) {
+        switch (AP_Filter::FilterType(params[i]._type)) {
+            case AP_Filter::FilterType::FILTER_NONE:
+                break;
+            case AP_Filter::FilterType::FILTER_NOTCH:
+                filters[i] = new AP_NotchFilter_params();
+                backend_var_info[i] = AP_NotchFilter_params::var_info;
+                count++;
+                break;
+            default:
+                return;
+        }
+
+        if (filters[i]) {
+            AP_Param::load_object_from_eeprom(filters[i], backend_var_info[i]);
+            AP_Param::invalidate_count();
+        }
+    }
+    if (enable == 0 && count > 0) {
```

Comment:
i think we can remove the enable parameter completely, reducing the number of steps to configure by 1

Reply:
Removed and added update